src/main/scala/com.snowplowanalytics.dataflow/streaming/StreamingCounts.scala

```diff
@@ -0,0 +1,181 @@
+import com.google.cloud.dataflow.sdk.transforms.PTransform
+import com.google.cloud.dataflow.sdk.transforms.GroupByKey
+import com.google.cloud.dataflow.sdk.transforms.windowing.FixedWindows
+import com.google.cloud.dataflow.sdk.transforms.windowing.Window
+import com.google.cloud.dataflow.sdk.values.KV
+import com.google.cloud.dataflow.sdk.values.PDone
+import com.google.cloud.dataflow.sdk.values.PCollection
+
+// Bigtable
+import com.google.cloud.bigtable.dataflow.AbstractCloudBigtableTableDoFn
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration
+import com.google.cloud.bigtable.dataflow.CloudBigtableTableConfiguration.Builder
+
+//HBase
+import org.apache.hadoop.hbase.client.Connection
+
+// This project
+import storage.BigtableUtils
+
+ /**
+   * transform applied to count the events per type in each bucket and 
+   * store the counts in Bigtable
+   */
+class StreamingCounts(config: CloudBigtableTableConfiguration, tableName: String, columnFamily: String)
+  extends PTransform[PCollection[KV[String, JIterable[String]]], PDone] {
+
+  override def apply(counts : PCollection[KV[String, JIterable[String]]]) : PDone = {
+  counts.apply(ParDo.named("StoreEventCounts").of(
+    StreamingCounts.storeEventCounts(config, tableName, columnFamily)))
+    PDone.in(counts.getPipeline)
+  }
+}
+
+object StreamingCounts {
+  /**
+   * Private function to set up Bigtable
+   *
+   * @param config The configuration for our job using StreamingCountsConfig.scala
+   * @return A Cloud Bigtable configuration instance with the options provided in config
+   */
+  private def setupBigtable(config: StreamingCountsConfig): CloudBigtableTableConfiguration =
+    new CloudBigtableTableConfiguration.Builder()
+      .withProjectId(config.projectId)
+      .withInstanceId(config.instanceId)
+      .withTableId(config.tableName)
+      .build()
+
+  /**
+   * Private function to set up Dataflow
+   *
+   * @param config The configuration for our job using StreamingCountsConfig.scala
+   * @param bigtableConfig The Bigtable config obtained from running setupBigtable
+   * @return a Pipeline instance, to which the transforms will be applied
+   */
+  private def setupDataflow(config: StreamingCountsConfig, 
+    bigtableConfig: CloudBigtableTableConfiguration): Pipeline = {
+ 
+    val options = PipelineOptionsFactory
+      .as(classOf[DataflowPipelineOptions])
+ 
+    options.setRunner(classOf[DataflowPipelineRunner])
+    options.setProject(config.projectId)
+    options.setStagingLocation(config.stagingLocation)
+    options.setStreaming(true)
+    Pipeline.create(options)
+  }
+
+  /**
+   * function applied in the "map" phase
+   */
+  def bucketEvents = new DoFn[String, KV[String, String]]() {
+
+    override def processElement(c: DoFn[String, KV[String, String]]#ProcessContext) : Unit = {
+      val e = SimpleEvent.fromJson(c.element)
+      if (e.isDefined) {
+        c.output(KV.of(e.get.bucket, e.get.`type`))
```

Comment:
```scala
val eventOpt = SimpleEvent.fromJson(c.element) // Option[SimpleEvent]
val kvOpt = eventOpt.map(e => KV.of(e.bucket, e.`type`)) // Option[KV[String, String]]
kvOpt.foreach(c.output) // If the option is defined the lambda in the foreach will be executed
```

It's good practice to avoid `get` on option even if you have an `isDefined` check above